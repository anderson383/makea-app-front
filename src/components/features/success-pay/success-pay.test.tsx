
import SuccessPay from '.'
import {render, fireEvent, waitFor} from '@testing-library/react-native'

const navigate = jest.fn()
const moduleRouter = {
  useNavigate: () => navigate
}
jest.mock('react-router-native', () => moduleRouter)

describe('Test SuccessPay Component', () => {
  
  const setup = async () => {
    return await waitFor(() => render(
      <SuccessPay />
    )) 
  }

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test(`Should render component`, async () => {
    const {toJSON} = await setup()

    expect(toJSON).toMatchSnapshot()
  })

  test(`Should click button buying`, async () => {
    const {getByTestId} = await setup()

    const button = getByTestId('styledButton')
    fireEvent.press(button)

    expect(navigate).toHaveBeenCalled()
  })
})