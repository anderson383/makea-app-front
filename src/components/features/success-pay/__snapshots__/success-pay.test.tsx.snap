// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SuccessPay Component Should render component 1`] = `[Function]`;
